src/upstream.py

```diff
@@ -19,9 +20,61 @@ def delete_upstream(upstream):
     shutil.rmtree(upstream, ignore_errors=True)
 
 
+def git_version():
```

Comment:
This function name suggests that it's returning a version number. I would expect it to return something like 2.35.3, but it's not.
Have you considered moving the `.replace(".", "")` part to `@unittest.skipUnless(2000 <= git_version(), 'needs git version > 2.0.0')`?